src/zjs_event.c

```diff
@@ -445,34 +433,27 @@ bool zjs_trigger_event(jerry_value_t obj,
                        zjs_post_event post,
                        void* h)
 {
-    event_t *ev;
     event_trigger_t *trigger = zjs_malloc(sizeof(event_trigger_t));
     if (!trigger) {
         ERR_PRINT("could not allocate trigger, out of memory\n");
         return false;
     }
 
-    int32_t callback_id = -1;
-
     jerry_value_t event_emitter = zjs_get_property(obj, HIDDEN_PROP("event"));
-    if (!jerry_get_object_native_handle(event_emitter, (uintptr_t*)&ev)) {
-        jerry_release_value(event_emitter);
-        zjs_free(trigger);
-        ERR_PRINT("native handle not found\n");
-        return false;
-    }
+    jerry_value_t map = zjs_get_property(event_emitter, "map");
+    jerry_value_t event_obj = zjs_get_property(map, event);
+    jerry_release_value(map);
     jerry_release_value(event_emitter);
 
-    jerry_value_t event_obj = zjs_get_property(ev->map, event);
     if (!jerry_value_is_object(event_obj)) {
-        jerry_release_value(event_obj);
```

Comment:
Don't remove this. you still need to free it - what if it was a string or error, etc. If OTOH you're already 100% sure it's an object, then don't check I guess.